go.mod

```diff
@@ -1,49 +1,72 @@
 module github.com/dell/csi-baremetal-operator
 
-go 1.16
+go 1.17
 
 require (
 	github.com/Masterminds/semver v1.5.0 // indirect
-	github.com/dell/csi-baremetal v0.2.2-beta
-	github.com/go-logr/logr v0.3.0
+	github.com/dell/csi-baremetal v0.5.0-alpha.0.20220602080507-ad5d19149ddb
+	github.com/go-logr/logr v1.2.0 // indirect
 	github.com/masterminds/semver v1.5.0
-	github.com/openshift/api v0.0.0-20200618202633-7192180f496a
-	github.com/sirupsen/logrus v1.6.0
-	github.com/stretchr/testify v1.5.1
+	github.com/openshift/api v0.0.0-20211214190212-1a67f3bd14ec
+	github.com/sirupsen/logrus v1.8.1
+	github.com/stretchr/testify v1.7.0
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
-	k8s.io/api v1.16.4
-	k8s.io/apimachinery v0.20.2
-	k8s.io/client-go v1.16.4
-	k8s.io/utils v0.0.0-20210111153108-fddb29f9d009
-	sigs.k8s.io/controller-runtime v0.7.2
-	sigs.k8s.io/controller-tools v0.5.0 // indirect
+	k8s.io/api v0.23.4
+	k8s.io/apimachinery v0.23.4
+	k8s.io/client-go v0.23.4
+	k8s.io/utils v0.0.0-20211116205334-6203023598ed
+	sigs.k8s.io/controller-runtime v0.11.1
```

Comment:
sigs.k8s.io/controller-tools missed, crd generation might be broken.

Reply:
thanks for pointing, `go build` replaced with `go install`

Reply:
It's better to keep it in the `go.mod`.

Reply:
`go mod` does not allow to keep `cmd` folder in imports - `Build constraints exclude all the Go files in '/home/user/go_projects/pkg/mod/sigs.k8s.io/controller-tools@v0.5.0/cmd'`
If we use any other path from `pkg` (for example: `sigs.k8s.io/controller-tools/pkg/version`) after `go mod tidy` all non related packages except ones from imported file are removed from `go.sum`. It will keep `sigs.k8s.io/controller-tools` in deps, but `go build` failed
```
GO111MODULE=on   go build -mod='' -o ./bin/ sigs.k8s.io/controller-tools/cmd/controller-gen
../../../../pkg/mod/sigs.k8s.io/controller-tools@v0.5.0/pkg/genall/help/pretty/help.go:9:2: missing go.sum entry for module providing package github.com/fatih/color (imported by sigs.k8s.io/controller-tools/pkg/genall/help/pretty); to add:
	go get sigs.k8s.io/controller-tools/pkg/genall/help/pretty@v0.5.0
../../../../pkg/mod/sigs.k8s.io/controller-tools@v0.5.0/pkg/crd/spec.go:23:2: missing go.sum entry for module providing package github.com/gobuffalo/flect (imported by sigs.k8s.io/controller-tools/pkg/crd); to add:
	go get sigs.k8s.io/controller-tools/pkg/crd@v0.5.0
../../../../pkg/mod/sigs.k8s.io/controller-tools@v0.5.0/cmd/controller-gen/main.go:25:2: missing go.sum entry for module providing package github.com/spf13/cobra (imported by sigs.k8s.io/controller-tools/cmd/controller-gen); to add:
	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0
../../../../pkg/mod/sigs.k8s.io/controller-tools@v0.5.0/pkg/loader/loader.go:30:2: missing go.sum entry for module providing package golang.org/x/tools/go/packages (imported by sigs.k8s.io/controller-tools/pkg/genall); to add:
```

So I prefer to leave controller-tools version in make variables

Reply:
Here is an example.
Create a directory `tools/tools.go` with code:

```
cat -p tools/tools.go
// +build tools

package tools

import (
    _ "sigs.k8s.io/controller-tools/cmd/controller-gen"
)
```

Next command `go mod tidy`:
```

$ go mod tidy
go: finding module for package sigs.k8s.io/controller-tools/cmd/controller-gen
go: downloading sigs.k8s.io/controller-tools v0.9.0
go: found sigs.k8s.io/controller-tools/cmd/controller-gen in sigs.k8s.io/controller-tools v0.9.0
go: downloading k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9

go: downloading k8s.io/klog/v2 v2.60.1
go: downloading github.com/spf13/cobra v1.4.0
go: downloading golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd
go: downloading github.com/gobuffalo/flect v0.2.5
go: downloading golang.org/x/tools v0.1.10-0.20220218145154-897bd77cd717
go: downloading golang.org/x/crypto v0.0.0-20210921155107-089bfa567519
```

the controller-tool is in the `go.mod`:

```diff
diff --git a/go.mod b/go.mod
index 8ac403d..8653448 100644
--- a/go.mod
+++ b/go.mod
@@ -5,18 +5,17 @@ go 1.16
 require (
        github.com/Masterminds/semver v1.5.0 // indirect
        github.com/dell/csi-baremetal v0.2.2-beta
-       github.com/go-logr/logr v0.3.0
        github.com/masterminds/semver v1.5.0
        github.com/openshift/api v0.0.0-20200618202633-7192180f496a
        github.com/sirupsen/logrus v1.6.0
-       github.com/stretchr/testify v1.5.1
+       github.com/stretchr/testify v1.7.0
        gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
        k8s.io/api v1.16.4
-       k8s.io/apimachinery v0.20.2
+       k8s.io/apimachinery v0.24.0
        k8s.io/client-go v1.16.4
-       k8s.io/utils v0.0.0-20210111153108-fddb29f9d009
+       k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
        sigs.k8s.io/controller-runtime v0.7.2
-       sigs.k8s.io/controller-tools v0.5.0 // indirect
+       sigs.k8s.io/controller-tools v0.5.0
 )

```

Reply:
all works, thanks! missed `// +build tools` annotation